File: NeuronMCP/internal/middleware/builtin/validation.go
package builtin

import (
	"context"

	"github.com/pgElephant/NeuronMCP/internal/middleware"
)

// ValidationMiddleware validates requests
type ValidationMiddleware struct{}

// NewValidationMiddleware creates a new validation middleware
func NewValidationMiddleware() *ValidationMiddleware {
	return &ValidationMiddleware{}
}

// Name returns the middleware name
func (m *ValidationMiddleware) Name() string {
	return "validation"
}

// Order returns the execution order
func (m *ValidationMiddleware) Order() int {
	return 1
}

// Enabled returns whether the middleware is enabled
func (m *ValidationMiddleware) Enabled() bool {
	return true
}

// Execute executes the middleware
func (m *ValidationMiddleware) Execute(ctx context.Context, req *middleware.MCPRequest, next middleware.Handler) (*middleware.MCPResponse, error) {
	if req.Method == "" {
		return &middleware.MCPResponse{
			Content: []middleware.ContentBlock{
				{Type: "text", Text: "Missing method in request"},
			},
			IsError: true,
		}, nil
	}

	if req.Params != nil {
		// Params should be a map, which is already validated by type
	}

	return next(ctx)
}

File: NeuronMCP/src/middleware/builtin/validation.ts
/**
 * Request validation middleware
 */

import { Middleware, MCPRequest, MCPResponse } from "../types.js";

export function createValidationMiddleware(): Middleware {
	return {
		name: "validation",
		order: 1,
		handler: async (request: MCPRequest, next: () => Promise<MCPResponse>): Promise<MCPResponse> => {
			// Basic validation
			if (!request.method) {
				return {
					content: [
						{
							type: "text",
							text: "Missing method in request",
						},
					],
					isError: true,
				};
			}

			// Validate params if present
			if (request.params && typeof request.params !== "object") {
				return {
					content: [
						{
							type: "text",
							text: "Invalid params: must be an object",
						},
					],
					isError: true,
				};
			}

			return next();
		},
	};
}





